### ISSUE TITLE
Some feedback

### ISSUE THREAD

**Issue by rasbt** [hugobowne/deep-learning-from-scratch-pytorch#5]
Hi Hugo,

here's some brief feedback for you :)

===

I think "backward propagation" goes nicely with "forward propagation", but the accepted technical term is "backpropagation" (the original long version from the 1986 Rumelhart & Hinton paper is "back-propagating errors")

> An optimization metric, such as accuracy.

This is what we often care about, but in the field, this is the "evaluation metric." The optimization metric (the surrogate for the evaluation metric) is usually log-likelihood/cross-entropy, information gain, hinge loss, some reward or fitness function etc.

> If your training data is biased, your algorithm more than likely will be;

I think you mean model instead of algorithm.

I would maybe include the code for making the tree figure. Some attendants may find it useful. You don't need GraphViz for that anymore, the plot_tree function in sklearn is based on matplotlib. See bottom of the notebook at https://github.com/rasbt/stat479-machine-learning-fs19/blob/master/06_trees/code/06-trees_demo_without_graphviz.ipynb 

I see that you switched to "back propogation" later :). Little nitpick: there is usually no whitespace, it's usually written as one word: backpropagation.

===

Overall, I really like the topic and tutorial flow. I am not sure how much time you have though. I think that you may have to be careful reg. the lengthy intro to machine learning. If this part becomes too long, at the expense of the rest, some attendees may be disappointed that you didn't get the to from scratch parts early enough. Personally, I would maybe cut the decision tree stuff and introduce logistic regression earlier. E.g., explaining it and applying it via the sklearn API first, and then implementing it from scratch. 

Also, I think that backpropagation is the hardest part for people who are new to it. So, I would definitely add more backpropagation-related figures. 

Otherwise, I think the tutorial is going to be in good shape after some polishing.

**Comment by hugobowne** [comment]
That's @rasbt, great feedback! 

I'll make sure to make those edits so that

- [x] the language I use is both correct and precise;
- [x] I reduce the amount of time spent on the intro to ML (I was actually thinking of skipping the linear regression -- just describing what it is and leaving it as an exercise for later);
- [x] there's more on backprop

to that end, do you have any favourite backprop figures?

these ones are cool! https://sebastianraschka.com/faq/docs/visual-backpropagation.html   ;)

**Comment by rasbt** [comment]
Regarding the backprop figure that would maybe be a good example. There's a little inaccuracy in that figure though; i have an updated version here: https://github.com/rasbt/python-machine-learning-book-3rd-edition/blob/master/ch12/images/12_12.png

But then it's maybe too mathematical to begin with. You could get rid of the formulas in that case and focus on explaining the flow of the "error terms" through the network to update the individual weight connections. 

Or, you could maybe work through a simpler example step by step (e.g., sth. like I did here, https://github.com/rasbt/stat453-deep-learning-ss20/blob/master/L06-pytorch/L06_pytorch__slides.pdf, slides 17-29)

![12_12](https://user-images.githubusercontent.com/5618407/86480018-dba05400-bd12-11ea-96ec-0f3193174fbd.png)

**Comment by hugobowne** [comment]
covered nearly everything here in https://github.com/hugobowne/deep-learning-from-scratch-pytorch/pull/6, except for `plot_tree function`, which I hope to include in a later iteration :)

thanks once again, @rasbt !
